File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(computers, visited, current):
    visited[current] = True
    for i in range(len(computers)):
        if computers[current][i] == 1 and not visited[i]:
            dfs(computers, visited, i)

def solution(n, computers):
    visited = [False] * n
    network_count = 0

    for i in range(n):
        if not visited[i]:
            dfs(computers, visited, i)
            network_count += 1

    return network_count

File: Java/프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        
        for (int i = 0; i < computers.length; i++) {
            if (!visited[i]) {
                dfs(computers, visited, i);
                answer += 1;
            }
        }
        return answer;
    }
    
    private void dfs(int[][] computers, boolean[] visited, int index) {
        visited[index] = true;
        
        for (int i = 0; i < computers.length; i++) {
            if (computers[index][i] == 1 && !visited[i]) {
                dfs(computers, visited, i);
            }
        }
    }
}